### PR TITLE
Re-implement job cancel functionality

### LINKED ISSUE #36: Reimplement cancel functionality
We can consolodate the cancel functionality of tooling jobs into 

- [ ] Remove Iteration::Representation::Cancel and Iteration::Analysis::Cancel
- [ ] Remove skipped tests
- [ ] Add a ToolingJob::Cancel command that updates the job_status to `cancelled`
- [ ] Update the relevant iteration's relevant status (`representation_status` or `analysis_status`)
  - (Maybe a `cancelled` state to the enum and try `analysis_cancelled!`)
  - (Check out the ToolingJob::Process command for inspiration)
- [ ] Add test coverage (obviously 😉)

### PR DESCRIPTION
**Pull request by ErikSchierboom**
Closes https://github.com/exercism/v3-website/issues/36﻿

### CODE REVIEW

File: app/commands/tooling_job/cancel.rb
Comment (iHiD) [inline app/commands/tooling_job/cancel.rb]: Rather than this, it should update the job's `job_status` in dynamodb to be `cancelled`. (This probably changed since we originally discussed this 🙂)

File: app/commands/iteration/test_run/process.rb
Comment (iHiD) [inline app/commands/iteration/test_run/process.rb]: So, in this scenario, there are actually two jobs that need cancelling - the analysis job and the representation job. So I think we need to call "ToolingJob::Cancel" twice with the two different `job_type`s.

**Review by iHiD — CHANGES_REQUESTED** [review]
Changes discussed 1-1 with Erik.

File: app/commands/tooling_job/cancel.rb
Comment (iHiD) [inline app/commands/tooling_job/cancel.rb]:
```suggestion
    initialize_with :iteration_uuid, :type
    
    def call
      config = ExercismConfig::SetupDynamoDBClient.()
```

**Comment by iHiD** [comment]
@ErikSchierboom I'm doing some work around tooling in general and wanted this to be green so that I could make sure I worked without giving you a horrible rebase to work from. Feel free to remove both my commits when you actually finish the PR 🙂

File: app/commands/tooling_job/cancel.rb
Comment (iHiD) [inline app/commands/tooling_job/cancel.rb]: If you drop the first `{` onto a new line I think this will indent less severely.

File: test/commands/tooling_job/cancel_test.rb
Comment (iHiD) [inline test/commands/tooling_job/cancel_test.rb]: Should this be unskipped?

File: test/commands/tooling_job/cancel_test.rb
Comment (iHiD) [inline test/commands/tooling_job/cancel_test.rb]: Should this be unskipped?

File: test/commands/tooling_job/cancel_test.rb
Comment (iHiD) [inline test/commands/tooling_job/cancel_test.rb]: Should this be unskipped?

File: test/commands/tooling_job/cancel_test.rb
Comment (iHiD) [inline test/commands/tooling_job/cancel_test.rb]: Should this be unskipped?

File: test/test_helper.rb
Comment (iHiD) [inline test/test_helper.rb:81]: Nice.

**Comment by iHiD** [comment]
Not sure if you intended the tests to all (but one) be skipping?

File: app/commands/tooling_job/cancel.rb
Comment (iHiD) [inline app/commands/tooling_job/cancel.rb]: Maybe move this into a method (`update_iteration`), and then the `iteration = Iteration.find_by!(uuid: iteration_uuid)` can come into this too?

File: app/commands/tooling_job/cancel.rb
Comment (iHiD) [inline app/commands/tooling_job/cancel.rb]: Maybe make this a memoized method called `items`?

File: app/commands/tooling_job/cancel.rb
Comment (iHiD) [inline app/commands/tooling_job/cancel.rb]: Then if you break this into `update_dymanodb`, you can have a call method like:
```
return unless items.length.positive?

update_dymamodb
update_iteration
```

**Comment by iHiD** [comment]
Also:
1. This is red
2. This will need the iteration -> submission rename.

I suggest rebasing and then doing (2). Sorry 🙂

**Comment by ErikSchierboom** [comment]
> Not sure if you intended the tests to all (but one) be skipping?

My only intention was for you to take a quick peek at the direction the PR was going in :)

File: scripts/setup_aws_locally.rb
Comment (ErikSchierboom) [inline scripts/setup_aws_locally.rb:111]: These are needed as otherwise we cannot filter on the `job_status` field nor return the `id` field (the default is only the key fields).

**Comment by ErikSchierboom** [comment]
@iHiD The tests are failing due to the new secondary index not being there. I assume that is because the [corresponding config PR](https://github.com/exercism/config/pull/10) has not yet been merged.

File: app/commands/tooling_job/cancel.rb
Comment (iHiD) [inline app/commands/tooling_job/cancel.rb]: I think this can just be `items.first`(ie no conditional needed).

You could even just put `.first` on the end of L70 and lose the `items` variable all together.

File: app/commands/tooling_job/cancel.rb
Comment (iHiD) [inline app/commands/tooling_job/cancel.rb]: Really minor comment, but helps with consistency. I tend to put these methods that don't do much at the bottom of the file, and then try to keep the other methods loosely in the order that they appear in the code.

So I'd have them ordered as:
```
pending_dynamodb_job
update_dynamodb
update_submission
client
```

**Review by iHiD — APPROVED** [review]
I've made a couple of stylistic comments but other than that it looks good! 🙂 

I'll approve, then once we've got it green, you can merge 😀

**Pull request merged**
